sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -126,9 +126,12 @@ object Cast {
    */
   def canUpCast(from: DataType, to: DataType): Boolean = (from, to) match {
     case _ if from == to => true
+    case (NullType, _) => false
```

Comment:
Is this covered by default case previously? Or is it missing before?

Reply:
This is because we need to match it before
```
case (_, StringType) => true
```